frontend_lib/test/mention.spec.js

```diff
@@ -0,0 +1,220 @@
+import { expect } from 'chai'
+import {
+  wrapMentionsInSpanTags,
+  addClassToMentionsOfUser,
+  removeClassFromMentionsOfUser,
+  handleMentionsBeforeSave
+} from '../src/mention.js'
+
+describe('mention.js', () => {
+  describe('function wrapMentionsInSpanTags', () => {
+    const parser = new global.DOMParser()
+
+    describe('with a source without any mention', () => {
+      const textWithoutMention = 'This is a text without any mention'
+      const document = parser.parseFromString(textWithoutMention, 'text/html')
+      wrapMentionsInSpanTags(document)
+      it('should not modify the source', () => expect(textWithoutMention).to.equal(document.body.innerHTML))
+    })
+
+    describe('with only one mention in the source', () => {
+      describe('with source as simple text', () => {
+        describe('with the mention at the middle of a sentence', () => {
+          const textWithMentionAtMiddle = 'This is a text with a mention @admin that should be wrapped'
+          const document = parser.parseFromString(textWithMentionAtMiddle, 'text/html')
+          wrapMentionsInSpanTags(document)
+          const addedSpanList = document.getElementsByTagName('span')
+          const addedSpanListId = addedSpanList[0].id
+
+          it('should only have one span tag', () => expect(addedSpanList).to.have.lengthOf(1))
+          it('should contain the username in the span tag', () => expect(addedSpanList[0].textContent).to.equal('@admin'))
+          it('should have the span id starting with "mention-"', () => expect(addedSpanListId.startsWith('mention-')).to.equal(true))
```

Comment:
(for all file) You can import the constants that you made at mention.js instead of use 'mention-', for example.